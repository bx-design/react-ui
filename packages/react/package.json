{
  "name": "@bx-design/react-ui",
  "version": "1.0.0",
  "sideEffects": false,
  "license": "MIT",
  "repository": "git@github.com:bx-design/react-ui.git",
  "author": "BX Creative Studio",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "typings": "dist/types/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/cjs/index.js",
      "default": "./dist/esm/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "lint": "yarn lint:src",
    "lint:src": "eslint src --ext .ts,.tsx --config ../../.eslintrc.json",
    "prebuild": "rimraf dist",
    "build": "concurrently yarn:build:*",
    "build:esm": "cross-env BABEL_ENV=esm babel src --root-mode upward --extensions .ts,.tsx -d dist/esm --source-maps",
    "build:cjs": "cross-env BABEL_ENV=cjs babel src --root-mode upward --extensions .ts,.tsx -d dist/cjs --source-maps",
    "build:types": "tsc -p ./tsconfig.json --declarationDir dist/types"
  },
  "dependencies": {
    "@bx-design/react-ui-alert": "^1.0.0",
    "@bx-design/react-ui-button": "^1.0.0",
    "@bx-design/react-ui-card": "^1.0.0",
    "@bx-design/react-ui-checkbox": "^1.0.0",
    "@bx-design/react-ui-dropdown": "^1.0.0",
    "@bx-design/react-ui-forms": "^1.0.0",
    "@bx-design/react-ui-modal": "^1.0.0",
    "@bx-design/react-ui-spinner": "^1.0.0",
    "@bx-design/react-ui-table": "^1.0.0",
    "@bx-design/react-ui-utils": "^1.0.0"
  },
  "devDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "peerDependencies": {
    "react": ">=17.0.2",
    "react-dom": ">=17.0.2"
  }
}
