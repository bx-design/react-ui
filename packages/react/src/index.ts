export * from '@bx-design/react-ui-alert'
export * from '@bx-design/react-ui-button'
export * from '@bx-design/react-ui-card'
export * from '@bx-design/react-ui-checkbox'
export * from '@bx-design/react-ui-dropdown'
export * from '@bx-design/react-ui-modal'
export * from '@bx-design/react-ui-spinner'
export * from '@bx-design/react-ui-table'
export * from '@bx-design/react-ui-forms'
export * from '@bx-design/react-ui-utils'
