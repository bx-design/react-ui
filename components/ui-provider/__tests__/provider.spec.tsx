import { render, screen } from '@testing-library/react'

import BxReactUiProvider from '../provider'
describe('<BxReactUiProvider />', () => {
  test('should render correctly', () => {
    render(
      <div>
        <BxReactUiProvider>
          <h1>Bx React Ui Provider</h1>
        </BxReactUiProvider>
      </div>,
    )
    expect(screen.getByText('Bx React Ui Provider')).toHaveTextContent(
      'Bx React Ui Provider',
    )
    expect(screen.getByText('Bx React Ui Provider')).toMatchSnapshot()
  })
})
