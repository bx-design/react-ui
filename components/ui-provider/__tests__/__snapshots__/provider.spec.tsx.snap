// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BxReactUiProvider /> should render correctly 1`] = `
<h1>
  Bx React Ui Provider
</h1>
`;
